services/spar/src/Spar/Error.hs

```diff
@@ -87,58 +99,64 @@ waiToServant waierr@(Wai.Error status label _) = ServantErr
   , errHeaders      = []
   }
 
-sparToWaiError :: SparError -> Either ServantErr Wai.Error
-sparToWaiError (SAML.CustomError SparNoSuchRequest)                       = Right $ Wai.Error status500 "server-error" "AuthRequest seems to have disappeared (could not find verdict format)."
-sparToWaiError (SAML.CustomError (SparNoRequestRefInResponse msg))        = Right $ Wai.Error status400 "server-error-unsupported-saml" ("The IdP needs to provide an InResponseTo attribute in the assertion: " <> msg)
-sparToWaiError (SAML.CustomError (SparCouldNotSubstituteSuccessURI msg))  = Right $ Wai.Error status400 "bad-success-redirect" ("re-parsing the substituted URI failed: " <> msg)
-sparToWaiError (SAML.CustomError (SparCouldNotSubstituteFailureURI msg))  = Right $ Wai.Error status400 "bad-failure-redirect" ("re-parsing the substituted URI failed: " <> msg)
-sparToWaiError (SAML.CustomError (SparBadInitiateLoginQueryParams label)) = Right $ Wai.Error status400 label label
-sparToWaiError (SAML.CustomError (SparBindFromWrongOrNoTeam msg))         = Right $ Wai.Error status403 "bad-team" ("Forbidden: wrong user team " <> msg)
-sparToWaiError (SAML.CustomError SparBindUserRefTaken)                    = Right $ Wai.Error status403 "subject-id-taken" "Forbidden: SubjectID is used by another wire user.  If you have an old user bound to this IdP, unbind or delete that user."
-
-sparToWaiError (SAML.CustomError (SparBadUserName msg))                   = Right $ Wai.Error status400 "bad-username" ("Bad UserName in SAML response, except len [1, 128]: " <> msg)
+renderSparErrorIO :: MonadIO m => Log.Logger -> SparError -> m (Either ServantErr Wai.Error)
```

Comment:
Again I think adding the suffix `IO` doesn't have any semantic info; We should have one function or the other; or at the very least rename `*IO` to `*WithLogging`

Reply:
`WithLogging` is a constructive suggestion; will do!  :)